### PR TITLE
Assigment 4 - Maria del Mar Sierra

### PR DESCRIPTION
**Pull request by Mariadelmarsierra**
Vuelvo a subir el assigment 4 ya que se eliminó al hacer el fork del repositorio del líder

### CODE REVIEW

File: Assignment4/Mariadelmarsierra-22c060/task07.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+# Visualize the results
+
+#for r in g.query(q1):
+#  print(r)
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+query_str = """
+SELECT ?subclass
+WHERE {
+    ?subclass rdfs:subClassOf ns:LivingThing .
```
Comment (dgarijo) [inline Assignment4/Mariadelmarsierra-22c060/task07.py:36]: You are not addressing the subclasses of subclasses of living thing. Hint: use `*`

File: Assignment4/Mariadelmarsierra-22c060/task07.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+"""
+
+# TO DO
+# Visualize the results
+query_str = """
+SELECT ?individual ?property ?value ?class
+WHERE {
+    ?individual a ?class .
+    FILTER (?class = ns:Person || ?class = ns:Animal)
```
Comment (dgarijo) [inline Assignment4/Mariadelmarsierra-22c060/task07.py:71]: very inefficient query. Use UNION instead

File: Assignment4/Mariadelmarsierra-22c060/task07.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    ?entity ns:knows ?knownEntity1 .
+    ?entity ns:knows ?knownEntity2 .
+    FILTER (STR(?knownEntity1) < STR(?knownEntity2))
+    BIND(CONCAT(STR(?knownEntity1), ", ", STR(?knownEntity2)) AS ?knownEntities)
+}
+"""
+q5 = prepareQuery(query_str, initNs={"ns": ns})
+results = g.query(q5)
+for row in results:
+  print(row)
```
Comment (dgarijo) [inline Assignment4/Mariadelmarsierra-22c060/task07.py:116]: Missing all exercises in rdflib

File: Assignment4/Mariadelmarsierra-22c060/task07.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+"""**Task 7.5: List the entities who know at least two other entities in the graph**"""
+
+# TO DO
+# Visualize the results
+query_str = """
+SELECT ?entity ?knownEntities
+WHERE {
+    ?entity a ?class .
+    FILTER (?class = ns:Person || ?class = ns:Animal)
```
Comment (dgarijo) [inline Assignment4/Mariadelmarsierra-22c060/task07.py:106]: this filter is unrelated to the problem

File: Assignment4/Mariadelmarsierra-22c060/task07.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    ?entity ns:knows ?knownEntity1 .
+    ?entity ns:knows ?knownEntity2 .
+    FILTER (STR(?knownEntity1) < STR(?knownEntity2))
```
Comment (dgarijo) [inline Assignment4/Mariadelmarsierra-22c060/task07.py:109]: why compare the strings? just do entity1!= entity2

File: Assignment4/Mariadelmarsierra-22c060/task07.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    BIND(CONCAT(STR(?knownEntity1), ", ", STR(?knownEntity2)) AS ?knownEntities)
```
Comment (dgarijo) [inline Assignment4/Mariadelmarsierra-22c060/task07.py:110]: note needed

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address the remaining changes

**Comment by anaigmo** [comment]
Entregado, corregido con algunos fallos, no se admiten más PR

**Pull request merged**